### PR TITLE
Fallback on `swift-frontend -scan-dependencies ...` fallback path in case `libSwiftScan` shared library cannot be loaded.

### PR DESCRIPTION
**Pull request by artemcm**
Resolves rdar://127768140

### CODE REVIEW

File: Sources/SwiftDriver/ExplicitModuleBuilds/ModuleDependencyScanning.swift [hunk 1]
```diff
@@ -171,12 +174,17 @@ public extension Driver {
...
-    try interModuleDependencyOracle.verifyOrCreateScannerInstance(fileSystem: fileSystem,
-                                                                  swiftScanLibPath: scanLibPath)
-    if isCachingEnabled {
-      self.cas = try interModuleDependencyOracle.getOrCreateCAS(pluginPath: try getCASPluginPath(),
-                                                                onDiskPath: try getOnDiskCASPath(),
-                                                                pluginOptions: try getCASPluginOptions())
+    do {
+      try interModuleDependencyOracle.verifyOrCreateScannerInstance(fileSystem: fileSystem,
+                                                                    swiftScanLibPath: scanLibPath)
+      if isCachingEnabled {
```
Comment (artemcm) [inline Sources/SwiftDriver/ExplicitModuleBuilds/ModuleDependencyScanning.swift:180]: @cachemeifyoucan should we error if caching is enabled but we failed to load libSwiftScan?

Comment (cachemeifyoucan) [reply Sources/SwiftDriver/ExplicitModuleBuilds/ModuleDependencyScanning.swift:180]: Error out sounds good to me.

**Review by cachemeifyoucan — COMMENTED** [review]
This only fixes broken libSwiftScan. I do believe if you want driver to load a broken scanner, you deserve a hard error with no fallback. Warning doesn't do anything in this case and user will assume they didn't do anything horrible if they don't hit any error afterwards.

**Comment by artemcm** [comment]
@swift-ci test

**Comment by artemcm** [comment]
@swift-ci test

**Review by cachemeifyoucan — APPROVED** [review]
Thanks. LGTM

**Review by owenv — APPROVED** [review]

**Pull request merged**